src/main/java/by/itech/lab/supplier/auth/CustomUserDetailsService.java

```diff
@@ -0,0 +1,49 @@
+package by.itech.lab.supplier.auth;
+
+import by.itech.lab.supplier.domain.User;
+import by.itech.lab.supplier.dto.CustomerDto;
+import by.itech.lab.supplier.dto.mapper.CustomerMapper;
+import by.itech.lab.supplier.repository.UserRepository;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service("userDetailsService")
+public class CustomUserDetailsService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+    private final CustomerMapper customerMapper;
+
+    public CustomUserDetailsService(UserRepository userRepository,
+                                    CustomerMapper customerMapper) {
+        this.userRepository = userRepository;
+        this.customerMapper=customerMapper;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userRepository.findByEmail(email);
+        UserImpl userImpl;
+        if (user == null) {
+            throw new UsernameNotFoundException(String.format("Username not found for email=%s", email));
+        } else {
+            List<CustomerDto> customers = new ArrayList<>();
+            customers.add(customerMapper.map(user.getCustomer()));
+            userImpl = new UserImpl(user.getUsername(), user.getPassword(), customers, true,
```

Comment:
if you're hardcoded default values for expiration and locks, maybe you can remove it from the default constructor and set them it in the UserImpl?

Reply:
not sure that understand what you want

Reply:
you always pass `boolean enabled, boolean accountNonExpired, boolean credentialsNonExpired` as true, is it correct?

Reply:
it's possibility add this functionality to user in the future

Reply:
so, when you will add this then you extend, for now, I see that its hardcoded and you do not necessary stuff

> there is nothing more permanent than temporary